### PR TITLE
meson: make version a dependency and use it in libbasic

### LINKED ISSUE #11483: Compilation fails intermittently due to missing "version.h"
**systemd version the issue has been seen with**
> latest master

**Used distribution**
 > Arch Linux

For some strange reason the systemd compilation fails intermittently (~twice in my last two dozen attempts) because the `version.h` file is generated too late in the compilation process:

```
    archlinux_systemd: [161/1762] Linking static target libjournal-core.a.
    archlinux_systemd: [162/1762] Compiling C object 'src/core/2ac6ece@@core@sta/job.c.o'.
    archlinux_systemd: [163/1762] Compiling C object 'systemd@exe/src_core_main.c.o'.
    archlinux_systemd: FAILED: systemd@exe/src_core_main.c.o 
    archlinux_systemd: cc -Isystemd@exe -I. -I.. -Isrc/basic -I../src/basic -Isrc/shared -I../src/shared -Isrc/systemd -I../src/systemd -Isrc/journal -I../src/journal -Isrc/journal-remote -I../src/journal-remote -Isrc/nspawn -I../src/nspawn -Isrc/resolve -I../src/resolve -Isrc/timesync -I../src/timesync -I../src/time-wait-sync -Isrc/login -I../src/login -Isrc/udev -I../src/udev -Isrc/libudev -I../src/libudev -Isrc/core -I../src/core -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -Isrc/libsystemd-network -I../src/libsystemd-network -I../ -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/uuid -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -g -Wextra -Werror=undef -Wlogical-op -Wmissing-include-dirs -Wold-style-definition -Wpointer-arith -Winit-self -Wfloat-equal -Wsuggest-attribute=noreturn -Werror=missing-prototypes -Werror=implicit-function-declaration -Werror=missing-declarations -Werror=return-type -Werror=incompatible-pointer-types -Werror=format=2 -Wstrict-prototypes -Wredundant-decls -Wmissing-noreturn -Wimplicit-fallthrough=5 -Wshadow -Wendif-labels -Wstrict-aliasing=2 -Wwrite-strings -Werror=overflow -Werror=shift-count-overflow -Werror=shift-overflow=2 -Wdate-time -Wnested-externs -Wno-unused-parameter -Wno-missing-field-initializers -Wno-unused-result -Wno-format-signedness -Wno-error=nonnull -Wno-maybe-uninitialized -ffast-math -fno-common -fdiagnostics-show-option -fno-strict-aliasing -fvisibility=hidden -fstack-protector -fstack-protector-strong --param=ssp-buffer-size=4 -Werror=shadow -include config.h -g -O0 -ftrapv -pthread  -MD -MQ 'systemd@exe/src_core_main.c.o' -MF 'systemd@exe/src_core_main.c.o.d' -o 'systemd@exe/src_core_main.c.o' -c ../src/core/main.c
    archlinux_systemd: In file included from ../src/core/main.c:27:
    archlinux_systemd: ../src/basic/build.h:4:10: fatal error: version.h: No such file or directory
    archlinux_systemd:  #include "version.h"
    archlinux_systemd:           ^~~~~~~~~~~
    archlinux_systemd: compilation terminated.
    archlinux_systemd: [164/1762] Compiling C object 'src/shared/5afaae1@@systemd-shared-240@sha/.._libudev_libudev-list.c.o'.
    archlinux_systemd: [165/1762] Compiling C object 'src/shared/5afaae1@@systemd-shared-240@sha/.._libudev_libudev-device.c.o'.
    archlinux_systemd: [166/1762] Compiling C object 'src/shared/5afaae1@@systemd-shared-240@sha/.._libudev_libudev-queue.c.o'.
    archlinux_systemd: [167/1762] Compiling C object 'src/shared/5afaae1@@systemd-shared-240@sha/.._libudev_libudev-enumerate.c.o'.
    archlinux_systemd: [168/1762] Compiling C object 'systemd-analyze@exe/src_analyze_analyze-verify.c.o'.
    archlinux_systemd: [169/1762] Compiling C object 'src/shared/5afaae1@@systemd-shared-240@sha/.._libudev_libudev.c.o'.
    archlinux_systemd: [170/1762] Generating ip-protocol-from-name.h with a meson_exe.py custom command.
    archlinux_systemd: [171/1762] Compiling C object 'src/shared/5afaae1@@systemd-shared-240@sha/.._libudev_libudev-util.c.o'.
    archlinux_systemd: [172/1762] Generating version.h with a custom command.
    archlinux_systemd: ninja: build stopped: subcommand failed.
```

Steps to "reproduce":
```
# pacman --noconfirm -Syu
## Install build dependencies
# pacman --needed --noconfirm -S acl cryptsetup docbook-xsl gperf lz4 xz pam libelf intltool \
    iptables kmod libcap libidn2 libgcrypt libmicrohttpd libxslt util-linux \
    linux-api-headers python-lxml quota-tools shadow gnu-efi-libs git meson \
    libseccomp pcre2 audit kexec-tools libxkbcommon bash-completion git ninja \
    gcc m4 pkgconf

# rm -fr build
# CFLAGS='-g -O0 -ftrapv' meson build \
      -Dslow-tests=true \
      -Dtests=unsafe \
      -Dinstall-tests=true \
      -Ddbuspolicydir=/etc/dbus-1/system.d \
      -Dnobody-user=nfsnobody \
      -Dnobody-group=nfsnobody
# ninja -C build
 ```

### PR DESCRIPTION
**Pull request by kragniz**
This should hopefully ensure it gets generated before basic build
happens.

Fixes #11483.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -580,6 +580,14 @@ vcs_tagger = [meson.source_root() + '/tools/meson-vcs-tag.sh',
               get_option('version-tag'),
               meson.project_version()]
 
+version_h = vcs_tag(
+        input : 'src/version/version.h.in',
+        output : 'version.h',
+        command: vcs_tagger,
+        fallback: meson.project_version())
```
Comment (keszybz) [inline meson.build]: This fallback line shouldn't be necessary, but looks good otherwise.

Comment (keszybz) [reply meson.build]: Yeah, please remove this line, since vcs-tagger already implements the fallback and there's no reason to duplicate it here.

**Comment by mrc0mmand** [comment]
I tried to recompile systemd in a Vagrant Arch VM ~10 times and the issue seems to be resolved. Thank you!

File: meson.build (hunk 1 above)
Comment (kragniz) [reply meson.build]: @keszybz removed

**Pull request merged**